code/modules/events/immovablerod.dm

```diff
@@ -52,16 +52,18 @@
 	locked_atoms = list()
 	grillepasschance = 0
 	mouse_opacity = 1
+	var/fuckMyShitUp = 0
 
-/obj/item/projectile/immovablerod/New(atom/start, atom/end)
+/obj/item/projectile/immovablerod/New(atom/start, atom/end, hyperRod = 0)
```

Comment:
The way you handle hyperrod is not OOP.
It should be a subtype, and the behaviour difference should be a proc, something like `clang_destroy(turf/T)`